src/platform/web/dom/History.js

```diff
@@ -65,6 +65,7 @@ export class History extends BaseObservableValue {
     }
 
     onSubscribeFirst() {
+        this._lastSessionHash = window.localStorage?.getItem("hydrogen_last_url_hash");
```

Comment:
Consider initializing _lastSessionHash to undefined in the constructor to not break [hidden classes](https://draft.li/blog/2016/12/22/javascript-engines-hidden-classes/).